ui.apps/src/main/content/jcr_root/apps/acs-commons/clientlibs/datepicker-with-timezone/js/datepicker.timezone.js

```diff
@@ -0,0 +1,171 @@
+            return;
+        }
+        var displayValue;
+        var $parent = $(this).parent();
+        var $datePicker = $parent.find("coral-datepicker");
+        var datePickerName = $datePicker.attr("name");
+        var datePickerValue = $datePicker.find('input[type="hidden"]').val();
+        var displayFormat = $datePicker.attr('displayformat');
+        var timeZoneName = datePickerName + "tz";
+        var timeZoneValue = $parent.find("coral-autocomplete[name='" + timeZoneName + "']").val();
+        if (!datePickerValue) {
+            return;
+        }
+        if (!moment(datePickerValue).isValid() && datePickerValue.indexOf('Z') !== -1) {
+            datePickerValue = datePickerValue.replace('Z', '');
+        }
+        if(displayFormat && displayFormat.indexOf("HH") && displayFormat.indexOf("mm")) {
+            if (timeZoneValue !== "Europe/London") {
+                var utcTime = moment.tz(datePickerValue, "Europe/London");
+                displayValue = utcTime.clone().tz(timeZoneValue).format(displayFormat);
+                $parent.find(".coral-InputGroup-input").val(displayValue);
+            } else {
+                if (datePickerValue.indexOf('Z') !== -1) {
+                    datePickerValue = datePickerValue.replace('Z', '');
+                }
+                displayValue = moment(datePickerValue).format(displayFormat);
+                $parent.find(".coral-InputGroup-input").val(displayValue);
+            }
+        }
+    });
+
+   var targetNodes = $("coral-datepicker");
+   targetNodes.each(function(index, targetNode){
+       if (targetNode !== undefined) {
+           if (targetNode.hasAttribute("disabled")) {
+               $(targetNode).parent().find(".datepickertz").attr("disabled", "disabled");
+           }
+           var config = {attributes: true, childList: false, subtree: false};
+           var callback = function (mutationsList, observer) {
+               mutationsList.forEach(function(mutation) {
+                   switch(mutation.type) {
+                       case 'attributes' :
+                           if (mutation.attributeName === "disabled") {
+                              if (mutation.target.hasAttribute("disabled")) {
+                                  $(mutation.target).parent().find(".datepickertz").attr("disabled", "disabled");
+                              } else {
+                                  $(mutation.target).parent().find(".datepickertz").removeAttr("disabled");
+                              }
+                           }
+                       break;
+                   }
+               });
+           };
+           // Create an observer instance linked to the callback function
+           var observer = new MutationObserver(callback);
+           // Start observing the target node for configured mutations
+           observer.observe(targetNode, config);
+       }
+   });
+
+    var initialLoad = 1;
+    initialLoadVar = true;
```

Comment:
You must declare this with `var` otherwise it will pollute the global namespace.